src/metabase/api/session.clj

```diff
@@ -282,14 +282,7 @@
 (api/defendpoint-schema GET "/properties"
   "Get all global properties and their values. These are the specific `Settings` which are meant to be public."
   []
-  (merge
-   (setting/user-readable-values-map :public)
-   (when @api/*current-user*
-     (setting/user-readable-values-map :authenticated))
-   (when (setting/has-advanced-setting-access?)
-     (setting/user-readable-values-map :settings-manager))
-   (when api/*is-superuser?*
-     (setting/user-readable-values-map :admin))))
+  (setting/user-readable-values-map (setting/current-user-readable-visibilities)))
```

Comment:
This refactoring extracts the logic from the `GET "/properties"` endpoint to the `setting/current-user-readable-visibilities` function. It was necessary to reuse the logic during the JSON encoding [here](https://github.com/metabase/metabase/pull/28150/files#diff-08f05088ade450091841832831efb413eb5204f0cd7eb1f253d5dfb9f7b50dc7R267-R273).